### ISSUE TITLE
Config.getConfig argument is misnamed

### ISSUE THREAD

**Issue by aparajita** [eslint/eslint#520]
Currently the signature of Config.getConfig is:

``` javascript
function(directory)
```

and the parameter comment is: "the directory to start looking for a local config".

In actual fact, Config.getConfig is called either with a path to a file or with no argument. In the former case, the parent directory of the file is where the search starts. In the latter case, the current working directory **should** be the starting point of the search.
